### ISSUE TITLE
Many Debugger Browser menu items raise errors.

### ISSUE THREAD

**Issue by refactoringdr** [guillep/pharo-core#10239]
**Bug description**

**To Reproduce**
1. Get a debugger (e.g. `3 zork`)
2. Right-click on the stack in the debugger select any of the following Browse commands:
 - method inheritance
 - method versions
 - implementors
 - senders
 - references
 - class references
 - class hierarchy
 - class variables
 - class variables references
 - instance variables references
3. Additionally: Explore->All Instances and Explore->all substances also crash.

**Expected behavior**
Appropriate browser is opened.

**Version information:**
 - OS: MacOS
 - Version: 11.6
 - Pharo Version 9.0

**Expected development cost**
As this version is marketed as "stable" I would have expected the fundamental tools and UI to have been tested during the considerable beta phase.

**Comment by MarcusDenker** [comment]
Checking the senders of buildSpecCommand:forContext:, the one that work hand over the debugger instance, the broken one hand over e.g. the context they ask from the debugger. 

It seems that a fix would be to make sure to hand over the debugger always and then change the commands to get the right information from it.

**Comment by maxwills** [comment]
I took a look too about this. As Markus said, the context field contains a Context object, when normally in these kind of commands, it should contain the specContext (in this case the instance of the StDebugger).

The context slot in the failing command has mixed semantics. The command "framework" expects it to have a presenter, and the implementation of this command subclasses expects them to have a Context object. This can be checked because the tests (which are passing) of this failing command expect a Context object.
The solution would be to add a different instance variable holding the target context (the Context object), and modify the tests accordingly.

I could not figure out how these command instantiation happens (It's kind of obscure to track) in the time I invested. If I manage to find more free minutes this week, I'll give this a shot.
